lib/tool_shed/util/shed_util_common.py

```diff
@@ -333,31 +336,34 @@ def handle_email_alerts(app, host, repository, content_alert_str='', new_repo_al
         else:
             email_from = 'galaxy-no-reply@' + host.split(':')[0]
         ctx = repo[repo.changelog.tip()]
+        username = unicodify(ctx.user())
         try:
-            username = ctx.user().split()[0]
+            username = username.split()[0]
         except Exception:
-            username = ctx.user()
+            pass
         # We'll use 2 template bodies because we only want to send content
         # alerts to tool shed admin users.
         if new_repo_alert:
             template = new_repo_email_alert_template
         else:
             template = email_alert_template
         display_date = hg_util.get_readable_ctx_date(ctx)
+        description = unicodify(ctx.description())
+        revision = '{}:{}'.format(ctx.rev(), ctx)
```

Comment:
do you know if `ctx.rev()` returns an integer or a bytestring ? If it's a bytestring we'd need to `unicodify(ctx.rev())`.

Reply:
OK, it's an integer.